internal/util/cache/lru_cache.go

```diff
@@ -143,8 +143,8 @@ func (c *LRU[K, V]) Add(key K, value V) {
 
 // Get returns value for provided key.
 func (c *LRU[K, V]) Get(key K) (value V, ok bool) {
-	c.m.RLock()
-	defer c.m.RUnlock()
+	c.m.Lock()
```

Comment:
we should probably add a separte lock for evictList.
but since LRUCache is only used for 2.3 for now. leave it